File: 9.Palindrome_Number.cpp
class Solution {
    public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        int original = x;
        long reversed  = 0;
        while ( x!=0){
            reversed = reversed*10 + (x%10);
            x /= 10;
            if (reversed > INT_MAX) return false;
        }
        reversed = static_cast<int>(reversed);
        if (reversed == original) return true;
        else return false;
    }
};

File: 9.Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int reverse = 0;
        int original = x;
        while (original != 0) {
            reverse = reverse*10 + original%10;
            original /= 10;
        }
        return reverse == x ? true : false;
    }
}